portsyncd/linksync.cpp

```diff
@@ -200,17 +197,12 @@ void LinkSync::onMsg(int nlmsg_type, struct nl_object *obj)
             return;
         }
 
+        /* Host interface is created */
         if (!g_init && g_portSet.find(key) != g_portSet.end())
         {
-            /* Bring up the front panel port as the first place*/
-            /* TODO: handle system retur code, if-block is just a warning suppressor */
-            if (system(("/sbin/ifup --force " + key).c_str()))
-                ;
             g_portSet.erase(key);
```

Comment:
the g_portSet is used to track all the ports that are needed for creating the host interfaces. If the host interfaces are created, in other words the nlmsg is recieved, the port is removed from the g_portSet. After all ports are removed from the g_portSet, all host interfaces are created and thus is the time to create the VLAN interfaces via kernel. Otherwise, the VLAN members cannot be added into the VLAN interface.
I remove the 'else' condition, so that after the host interface is created, the nlmsg will be sent to the db via the set function. This is the place that the port admin status is set.